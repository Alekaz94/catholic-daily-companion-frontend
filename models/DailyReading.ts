export interface DailyReading {
    id: string;
    createdAt: string;
    firstReading: string;
    secondReading: string | null;
    psalm: string;
    gospel: string;
}

export interface NewDailyReading {
    firstReading: string;
    secondReading?: string;
    psalm: string;
    gospel: string;
}