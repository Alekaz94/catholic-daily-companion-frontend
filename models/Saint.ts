export interface Saint {
    id: string;
    name: string;
    birthYear: number;
    deathYear: number;
    feastDay: string | null;
    biography: string;
    patronage: string;
    canonizationYear: number;
    imageUrl: string | null;
}

export interface NewSaint {
    name: string;
    birthYear: number;
    deathYear: number;
    feastDay: string;
    biography: string;
    patronage: string;
    canonizationYear: number;
    imageUrl: string;
}