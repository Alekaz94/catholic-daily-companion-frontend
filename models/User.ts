export interface User {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  role: string;
}

export interface NewUser {
  firstName: string;
  lastName: string;
  email: string;
  password: string;
}

export interface UserPasswordChange {
  password: string;
}
