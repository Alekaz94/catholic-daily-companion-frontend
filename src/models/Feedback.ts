export interface Feedback {
    id: string;
    category: string;
    message: string;
    email?: string;
    submittedAt?: string;
}

export interface FeedbackRequest {
    category: string;
    message: string;
    email?: string;
}