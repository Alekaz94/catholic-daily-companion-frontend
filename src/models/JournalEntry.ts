export interface JournalEntry {
    id: string;
    date: string;
    updatedAt: string;
    title: string;
    content: string;
}

export interface NewJournalEntry {
    title: string;
    content: string;
}

export interface UpdateJournalEntry {
    title?: string;
    content?: string;
}