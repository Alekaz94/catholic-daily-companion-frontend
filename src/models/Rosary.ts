export interface Rosary {
    id: string;
    date: string;
    completed: boolean;
}