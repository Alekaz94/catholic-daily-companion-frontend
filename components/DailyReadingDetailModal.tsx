import React from "react";
import { Modal, View, Text, StyleSheet, Button } from "react-native";
import { DailyReading } from "../models/DailyReading";

interface Props {
    visible: boolean;
    reading: DailyReading | null;
    onClose: () => void;
}

const DailyReadingDetailModal: React.FC<Props> = ({visible, reading, onClose}) => {
    if(!reading) {
        return null;
    }

    return (
        <Modal visible={visible} animationType='slide'>
            <View style={styles.modalView}>
                <Text style={styles.modalTitle}>Today's readings</Text>
                <Text style={styles.modalDate}>{reading.createdAt}</Text>
                <Text style={styles.modalReadings}>First reading: {reading.firstReading}</Text>
                <Text style={styles.modalReadings}>Second reading: {reading.secondReading}</Text>
                <Text style={styles.modalReadings}>Psalm: {reading.psalm}</Text>
                <Text style={styles.modalReadings}>Gospel reading: {reading.gospel}</Text>
            
                <Button title="Close" onPress={onClose} />
            </View>
        </Modal>
    );
}

const styles = StyleSheet.create({
    modalView: {
        flex: 1, 
        padding: 20
    },
    modalTitle: {
        fontSize: 20, 
        fontWeight: "bold"
    },
    modalDate: {
        marginVertical: 10, 
        fontSize: 10
    },
    modalReadings: {
        marginBottom: 16,
        padding: 10
    }
})

export default DailyReadingDetailModal;