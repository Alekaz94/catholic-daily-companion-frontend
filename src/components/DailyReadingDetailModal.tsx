import React from "react";
import { Modal, View, Text, TouchableOpacity } from "react-native";
import { DailyReading } from "../models/DailyReading";
import { Layout } from "../styles/Layout";
import { Typography } from "../styles/Typography";
import { AppTheme } from "../styles/colors";

interface Props {
    visible: boolean;
    reading: DailyReading | null;
    onClose: () => void;
}

const DailyReadingDetailModal: React.FC<Props> = ({visible, reading, onClose}) => {
    if(!reading) {
        return null;
    }

    return (
        <Modal visible={visible} animationType='slide'>
            <View style={[Layout.container, {backgroundColor: AppTheme.reading.background}]}>
                <Text style={[Typography.title, {color: AppTheme.reading.text}]}>Today's readings</Text>
                <Text style={[Typography.small, {color: AppTheme.reading.text}]}>{reading.createdAt}</Text>
                <Text style={[Typography.label, {marginBottom: 5, marginTop: 20, color: AppTheme.reading.text, borderBottomWidth: 1}]}>First reading </Text>
                <Text style={[Typography.body, {color: AppTheme.reading.text}]}>{reading.firstReading}</Text>
                {reading.secondReading ? 
                    <> 
                        <Text style={[Typography.label, {marginBottom: 5, marginTop: 20, borderBottomWidth: 1, color: AppTheme.reading.text}]}>Second reading </Text>
                        <Text style={[Typography.body, {color: AppTheme.reading.text}]}>{reading.secondReading}</Text>
                    </>
                    : 
                    <>
                    </>
                }
                <Text style={[Typography.label, {marginBottom: 5, marginTop: 20, borderBottomWidth: 1, color: AppTheme.reading.text}]}>Psalm </Text>
                <Text style={[Typography.body, {color: AppTheme.reading.text}]}>{reading.psalm}</Text>
                <Text style={[Typography.label, {marginBottom: 5, marginTop: 20, borderBottomWidth: 1, color: AppTheme.reading.text}]}>Gospel reading </Text>
                <Text style={[Typography.body, {marginBottom: 20, color: AppTheme.reading.text}]}>{reading.gospel}</Text>
            
                <TouchableOpacity onPress={onClose} style={[Layout.button, {width: "50%", alignSelf: "center", backgroundColor: "#ADD8E6", borderWidth: 1}]}>
                    <Text style={[Layout.buttonText, {alignSelf: "center", color: AppTheme.reading.text}]}>Close</Text>
                </TouchableOpacity> 
            </View>
        </Modal>
    );
}

export default DailyReadingDetailModal;