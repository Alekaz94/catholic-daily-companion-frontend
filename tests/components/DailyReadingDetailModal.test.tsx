import { act, fireEvent, render } from "@testing-library/react-native";
import DailyReadingDetailModal from "../../components/DailyReadingDetailModal";
import { DailyReading } from "../../models/DailyReading";

const mockReading: DailyReading = {
    id: "string",
    createdAt: "2025-05-16",
    firstReading: "Exodus",
    secondReading: "Hebrews",
    psalm: "Psalm 123",
    gospel: "John"
};

describe("DailyReadingDetailModal", () => {
    it("does not render if daily reading is null", () => {
        const { toJSON } = render(
            <DailyReadingDetailModal visible={true} reading={null} onClose={jest.fn()} />
        );
        expect(toJSON()).toBeNull();
    })

    it("does not render content if visible is false", () => {
        const { queryByText } = render(
            <DailyReadingDetailModal visible={false} reading={null} onClose={jest.fn()} />
        );
    
        expect(queryByText("Test entry")).toBeNull();
    });

    it("renders daily reading data correctly", () => {
        const { getByText } = render(
            <DailyReadingDetailModal visible={true} reading={mockReading} onClose={jest.fn()} />
        )
        expect(getByText("Today's readings")).toBeTruthy();
        expect(getByText("2025-05-16")).toBeTruthy();
        expect(getByText("First reading")).toBeTruthy();
        expect(getByText("Exodus")).toBeTruthy();
        expect(getByText("Second reading")).toBeTruthy();
        expect(getByText("Hebrews")).toBeTruthy();
        expect(getByText("Psalm")).toBeTruthy();
        expect(getByText("Psalm 123")).toBeTruthy();
        expect(getByText("Gospel reading")).toBeTruthy();
        expect(getByText("John")).toBeTruthy();
    })

    it("calls onClose when Close button is pressed", async () => {
        const onCloseMock = jest.fn();
        const { getByText } = render(
            <DailyReadingDetailModal visible={true} reading={mockReading} onClose={onCloseMock} />
        )
        
        await act(async () => {
            fireEvent.press(getByText("Close"));
        })
        expect(onCloseMock).toHaveBeenCalled();
    })

    it("renders no second reading if second reading is null", () => {
        const reading = {...mockReading, secondReading: null};
        const { queryByText } = render(
            <DailyReadingDetailModal visible={true} reading={reading} onClose={jest.fn()} />
        )
        expect(queryByText("Second reading")).toBeNull();
        expect(queryByText("First reading")).toBeTruthy();
        expect(queryByText("Psalm")).toBeTruthy();
        expect(queryByText("Gospel reading")).toBeTruthy();
    })

    it("can handle multiple close button presses", async () => {
        const onCloseMock = jest.fn();
        const { getByText } = render(
            <DailyReadingDetailModal visible={true} reading={mockReading} onClose={onCloseMock} />
        );
    
        const closeButton = getByText("Close");
    
        await act(async () => {
            fireEvent.press(closeButton);
            fireEvent.press(closeButton);
        });
    
        expect(onCloseMock).toHaveBeenCalledTimes(2);
    });

    it("matches the snapshot", () => {
        const { toJSON } = render(
            <DailyReadingDetailModal visible={true} reading={mockReading} onClose={jest.fn()} />
        );
        expect(toJSON()).toMatchSnapshot();
    });
})