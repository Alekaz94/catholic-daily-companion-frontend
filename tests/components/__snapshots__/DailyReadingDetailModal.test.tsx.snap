// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyReadingDetailModal matches the snapshot 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      [
        {
          "flex": 1,
          "padding": 20,
        },
        {
          "backgroundColor": "#F0F9FF",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#222",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 16,
          },
          {
            "color": "#1A1A1A",
          },
        ]
      }
    >
      Today's readings
    </Text>
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 12,
          },
          {
            "color": "#1A1A1A",
          },
        ]
      }
    >
      2025-05-16
    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 14,
            "fontWeight": "500",
            "marginBottom": 4,
          },
          {
            "alignSelf": "center",
            "color": "#1A1A1A",
            "marginBottom": 5,
            "marginTop": 20,
          },
        ]
      }
    >
      First reading 
    </Text>
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 14,
          },
          {
            "color": "#1A1A1A",
          },
        ]
      }
    >
      Exodus
    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 14,
            "fontWeight": "500",
            "marginBottom": 4,
          },
          {
            "alignSelf": "center",
            "color": "#1A1A1A",
            "marginBottom": 5,
            "marginTop": 20,
          },
        ]
      }
    >
      Second reading 
    </Text>
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 14,
          },
          {
            "color": "#1A1A1A",
          },
        ]
      }
    >
      Hebrews
    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 14,
            "fontWeight": "500",
            "marginBottom": 4,
          },
          {
            "alignSelf": "center",
            "color": "#1A1A1A",
            "marginBottom": 5,
            "marginTop": 20,
          },
        ]
      }
    >
      Psalm 
    </Text>
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 14,
          },
          {
            "color": "#1A1A1A",
          },
        ]
      }
    >
      Psalm 123
    </Text>
    <Text
      style={
        [
          {
            "color": "black",
            "fontSize": 14,
            "fontWeight": "500",
            "marginBottom": 4,
          },
          {
            "alignSelf": "center",
            "color": "#1A1A1A",
            "marginBottom": 5,
            "marginTop": 20,
          },
        ]
      }
    >
      Gospel reading 
    </Text>
    <Text
      style={
        [
          {
            "color": "white",
            "fontSize": 14,
          },
          {
            "color": "#1A1A1A",
            "marginBottom": 20,
          },
        ]
      }
    >
      John
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#ADD8E6",
          "borderRadius": 8,
          "marginTop": 10,
          "opacity": 1,
          "paddingVertical": 12,
          "width": "50%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
            },
            {
              "alignSelf": "center",
              "color": "#1A1A1A",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</Modal>
`;
